File: android/app/src/main/java/com/wikiart/PaintingList.kt
package com.wikiart

import com.google.gson.annotations.SerializedName

data class PaintingList(
    @SerializedName("Paintings") val paintings: List<Painting>,
    @SerializedName("AllPaintingsCount") val allPaintingsCount: Int,
    @SerializedName("PageSize") val pageSize: Int
) {
    val pageCount: Int
        get() = (allPaintingsCount + pageSize - 1) / pageSize
}

File: WikiArt3.0/Models/PaintingList.swift
//
//  Painting.swift
//  Sticky Art
//
//  Created by Waleed Azhar on 2019-08-09.
//  Copyright © 2019 Waleed Azhar. All rights reserved.
//

import Foundation

class PaintingList: Codable {
    var paintings: [Painting]
    let allPaintingsCount, pageSize: Int
    
    enum CodingKeys: String, CodingKey {
        case paintings = "Paintings"
        case allPaintingsCount = "AllPaintingsCount"
        case pageSize = "PageSize"
    }
    
    init(list: [Painting]) {
        self.pageSize = 1
        self.paintings = list
        self.allPaintingsCount = paintings.count
    }
    
}

extension PaintingList {
    
    var pageCount: Int {
        return (allPaintingsCount + pageSize - 1) / pageSize
    }
    
}
